### PR TITLE
OpenAPI: comments_api

### PR DESCRIPTION
**Pull request by markxiong0122**
implemented comments_api with OAS

### CODE REVIEW

File: api/comments_api.py
Comment (markxiong0122) [inline api/comments_api.py:128]: @jrobbins @DanielRyanSmith Do we want to do such parsing to cast it to the auto-generated model or just keep what it was previously?

Comment (jrobbins) [reply api/comments_api.py:128]: Yes, I'd like to see us use the auto-generated model for the request parameters.

File: api/comments_api.py
Comment (jrobbins) [inline api/comments_api.py:97]: I looked into the generated controller and model classes and it looks like the `from_dict()` method is the easiest way to have the generated code validate the input.   So, could these 6 lines be just one line?
`comment_request = CommentRequest.from_dict(self.request.json)`

File: api/comments_api.py
Comment (jrobbins) [inline api/comments_api.py:35]: This still seems to return `dict[str, Any]` or more precisely `dict[str, str]`.  You could probably make it return `AmendmentModel` and I think that would work with `ActivityModel`.

File: api/comments_api.py
Comment (markxiong0122) [reply api/comments_api.py:97]: this works and is a lot easier! Thanks

File: api/comments_api.py
Comment (markxiong0122) [reply api/comments_api.py:35]: That would work but requires changing the test file too. I'll push some changes and see if that is what we want

**Review by jrobbins — APPROVED** [review]
LGTM after considering one comment.

File: api/comments_api.py
Comment (jrobbins) [inline api/comments_api.py:83]: #4213 has merged, so I think that this can return the `GetCommentResponse` directly without calling to_dict().

**Review by jrobbins — APPROVED** [review]
One more suggestion

File: api/comments_api.py
Comment (markxiong0122) [reply api/comments_api.py:83]: The dict conversion does not work and still gives such errors when removing it `======================================================================
ERROR: test_patch__undelete_comment (api.comments_api_test.CommentsAPITest.test_patch__undelete_comment)
Handler unmarks a comment as deleted as requested by authorized user.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mark/chromium-dashboard/api/comments_api_test.py", line 249, in test_patch__undelete_comment
    self.assertEqual(get_resp['comments'][0]['deleted_by'], None)
                     ~~~~~~~~^^^^^^^^^^^^
TypeError: 'GetCommentsResponse' object is not subscriptable

======================================================================
ERROR: test_post__comment_only (api.comments_api_test.CommentsAPITest.test_post__comment_only)
Handler adds a comment only, does not require approval permission.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.11/3.11.8/Frameworks/Python.framework/Versions/3.11/lib/python3.11/unittest/mock.py", line 1375, in patched
    return func(*newargs, **newkeywargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mark/chromium-dashboard/api/comments_api_test.py", line 268, in test_post__comment_only
    self.assertEqual(actual, {'message': 'Done'})
  File "/opt/homebrew/Cellar/python@3.11/3.11.8/Frameworks/Python.framework/Versions/3.11/lib/python3.11/unittest/case.py", line 873, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/opt/homebrew/Cellar/python@3.11/3.11.8/Frameworks/Python.framework/Versions/3.11/lib/python3.11/unittest/case.py", line 863, in _baseAssertEqual
    if not first == second:
           ^^^^^^^^^^^^^^^
  File "/Users/mark/chromium-dashboard/cs-env/lib/python3.11/site-packages/chromestatus_openapi/models/base_model.py", line 64, in __eq__
    return self.__dict__ == other.__dict__
                            ^^^^^^^^^^^^^^
AttributeError: 'dict' object has no attribute '__dict__'` It looks like in the test file it still expect the response to be returned by the do_get method defined in the specific handler

Comment (jrobbins) [reply api/comments_api.py:83]: Sure, the tests would need to be updated to validate the returned model object instead of a returned dict.  It's OK with me if that is left for a future PR.

**Pull request merged**
